### PR TITLE
Initial version of CACCL-api-forwarder

### PR DESCRIPTION
**Pull request by gabeabrams**
The purpose of this module is to add functionality to an express.js server so the server can forward Canvas API requests from the client (react, angular, etc) to Canvas. 

Big picture of the process we're doing:

- Listen for new requests from the client (allow a prefix before the forwarded path so the server knows that all forwarded routes are grouped within a subpath)
- Upon receiving a new request, add credentials (access token) and send the request
- Once Canvas responds, forward the response to the client

The main file you'll need to look at is `index.js`

### CODE REVIEW

**Review by phoebem — APPROVED** [review]
Looks good

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,58 @@
+{
+  "extends": "airbnb",
```
Comment (lbjay) [inline .eslintrc.json:2]: curious, what exactly happens when you do this? does eslint go out and fetch the airbnb rules, or is it some baked-in profile?

Comment (lbjay) [reply .eslintrc.json:2]: oh haha i see it in the package.json dependencies. nevermind!

File: index.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    // Send the request
+    sendRequest({
+      host,
+      path,
+      numRetries,
+      method: req.method,
+      params: data,
+    })
+      .then((response) => {
+        res.json(response.body);
```
Comment (lbjay) [inline index.js]: is this all that's ever needed? would the canvas response ever include headers that would be necessary or useful to a client app?

**Review by lbjay — APPROVED** [review]

File: index.js (hunk 2 above)
Comment (gabeabrams) [reply index.js]: Great feedback. I'm also forwarding status code now. So far, we haven't needed headers for API calls. Other types of requests have needed headers, but those are not forwarded by this code.

File: .eslintrc.json (hunk 1 above)
Comment (gabeabrams) [reply .eslintrc.json:2]: It's usually a safe bet to expect that everything's a node module :)

**Pull request closed without merging**